docs/rfc/rfc-1.adoc

```diff
@@ -0,0 +1,143 @@
+:toc: macro
+
+= RFC 1: tBTCv2 Design
+
+:icons: font
+:numbered:
+toc::[]
+
+== Overview
+
+=== Minting
+
+Stakers periodically create 51-of-100 ecdsa-backed wallets to hold frozen BTC
+assets to maintain the peg. Depositors send BTC funds to the
+most-recently-created-wallet by using pay-to-script-hash (P2SH) which contains
+hashed information about the depositor's minting ethereum address. Once
+the block is mined, the depositor reveals their desired ethereum minting
+address to the ethereum chain. The TBTC system listens for these sorts of
+messages and when it gets one, it checks the bitcoin network to make sure the
+funds line up. If everything checks out, then we mint!
+
+=== Redemption
+
+A user supplies tBTCv2 and a bitcoin address. The system burns the tBTCv2 and
+releases the equivalent amount of bitcoin to the user from the oldest
+outstanding bitcoin wallet. If the oldest does not have enough, the remaining
+balance carries over to the next oldest until an equivalent amount of BTC has
+been released.
+
+=== Continuous Fees
+
+The system is able to collect fees in the form of tBTC when the BTC is
+deposited and when it is redeemed. Rather than paying operators at those two
+instances we would prefer to pay them continuously. Since we do not have the
+tBTC to pay them with until we collect the redemption fee (and can't mint it
+and maintain the peg), we must turn to another token. Instead, the system pays
+operators continuously in mintable T tokens and whenever it collects tBTC fees
+it uses those fees to buy T tokens back for its treasury.
+
+== In Depth
+
+=== Minting
+
+==== Deposit + Commitment
+
+Once we know the active wallet's public key hash, the dApp can put together a
+pay-to-script-hash (P2SH) address to receive the funds. This script will be
+unique to each depositor and will look like:
+
+```
+<eth-address> DROP
+<blinding-factor> DROP
+DUP HASH160 <signingGroupPubkey> EQUAL
+IF
+  CHECKSIG
+ELSE
+  DUP HASH160 <refundPubkey> EQUALVERIFY
+  <locktime> CHECKLOCKTIMEVERIFY DROP
+  CHECKSIG
+ENDIF
+```
+
+Since each depositor has their own ethereum address and their own secret
+blinding factor (which is an additional security layer), each depositor's
```

Comment:
> I'd love to see an explanation of why the blinding factor is needed here. Is it because we need to strengthen the hash? 

Yeah - the basic idea is that if a bad actor gets in control of the wallet, they can take the btc funds without the user revealing their eth address by grinding on every active eth address. @mhluongo has a better handle on the attack he has in mind though!